demos/common/cpp/utils/include/utils/images_capture.h

```diff
@@ -15,6 +15,7 @@ class ImagesCapture {
     ImagesCapture(bool loop) : loop{loop} {}
     virtual double fps() const = 0;
     virtual cv::Mat read() = 0;
+    virtual cv::Size getFrameSize() = 0;
```

Comment:
Is it possible to expand interface? Should i find way without changes to "common" interfaces?

Reply:
Well, it is, but what for?
You can add an explanation to get an answer or wait when I remember what was going in the PR.

Reply:
I think I get it :) `VideoCapture` can't provide values for sizes sometimes. For example if you give it % based pattern. You may get {0, 0} from `getFrameSize()`.

Is the a way to read the first frame manually, build a graph and feed it first with that frame and then with the rest of VideoCapture?

Reply:
I need to get size of frame for adjust some elements before start of graph. Part of them i can create based on first frame from graph (VideoWriter for example). But before starting graph, it is necessary to configure reshape of face detector.

Reply:
I can `cap->read()` first frame, get its size and then `reset` cap ptr to source start. Is it acceptable solution?

Reply:
Alternatively we could read the first frame once an input is opened and return that in `getFrameSize()` but It won't be guaranteed that all next frames are of the same resolution. Currently newer reshape dependent models warn about resolution change: https://github.com/openvinotoolkit/open_model_zoo/blob/1f5d596061a5199cc1a7cf86ec8af9653dc693af/demos/common/python/models/hpe_associative_embedding.py#L84
So I don't think it is a good idea to provide a solution that would hide the problem providing resolution of the first frame.

I think your solution is the best we can, so yes.

Reply:
Added this crutch.